From 5e57b2bcb420f683fcf9e0cc94a1bf104cd3c529 Mon Sep 17 00:00:00 2001
From: Fugang Duan <fugang.duan@nxp.com>
Date: Fri, 26 May 2017 14:09:27 +0800
Subject: [PATCH 1875/5242] MLK-15031-01 tty: serial: fsl_lpuart: drop the
 error frame

commit  f33e606acb91233814be682e29d33f54a617c127 from
https://source.codeaurora.org/external/imx/linux-imx.git

Since the driver stats the error frames in port.icount.frame, it can
drop the error frame that no need to push it to the tty buffer.

Signed-off-by: Fugang Duan <fugang.duan@nxp.com>
Reviewed-by: Robin Gong <yibin.gong@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/tty/serial/fsl_lpuart.c |    1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/tty/serial/fsl_lpuart.c b/drivers/tty/serial/fsl_lpuart.c
index 70b7e0e..608a5f3 100644
--- a/drivers/tty/serial/fsl_lpuart.c
+++ b/drivers/tty/serial/fsl_lpuart.c
@@ -863,6 +863,7 @@ static irqreturn_t lpuart32_rxint(int irq, void *dev_id)
 #ifdef SUPPORT_SYSRQ
 			sport->port.sysrq = 0;
 #endif
+			continue;
 		}
 
 		if (rx & UARTDATA_INVALID)
-- 
1.7.9.5

